external-table/src/pxfheaders.c

```diff
@@ -459,32 +546,34 @@ add_projection_desc_httpheader(CHURL_HEADERS headers,
 						 attrNumber + 1 - FirstLowInvalidHeapAttributeNumber);
 	}
 
+	// STEP 4: add projection headers for attributes of relation that are not dropped and were selected in steps before
```

Comment:
not sure if this is the best rewording, but it is personally easier for me to parse the following: 
```suggestion 
	// STEP 4: for attributes in the relation that are not dropped, add projection headers for those selected in steps 0 - 3 above